tests/integration/test_integration.py

```diff
@@ -175,7 +175,7 @@ async def test_when_scale_app_beyond_1_then_only_one_unit_is_active(
     )
     unit_statuses = Counter(unit.workload_status for unit in app.units)
     assert unit_statuses.get("active") == 1
-    assert unit_statuses.get("blocked") == 2
+    assert unit_statuses.get("unknown") == 2
```

Comment:
Hmmm, that's not good, isn't it? The status should actually be `Blocked` in this case. 
In the charm, if we return from the constructor right away, is `CollectStatusEvent` fired?

Reply:
According to documentation https://ops.readthedocs.io/en/latest/#ops.CollectStatusEvent  collect_app_status will only be triggered on the leader unit. That explains the reason.
Now the question is how could we set a status for non-leader units ?

Reply:
In the constructor - just as we used to do :)

Reply:
Found the solution. Registering collect_status observer before we return in the constructor should fix the issue.